src/reformat_libyuv.c

```diff
@@ -788,208 +516,113 @@ avifResult avifImageYUVToRGBLibYUV10bpc(const avifImage * image,
-                                 image->width,
-                                 image->height) != 0) {
-                return AVIF_RESULT_REFORMAT_FAILED;
-            }
-#endif
-            return AVIF_RESULT_OK;
-        } else if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV400) {
-            // This doesn't currently exist in libyuv
-        }
-    } else if (rgb->format == AVIF_RGB_FORMAT_RGBA) {
-        // AVIF_RGB_FORMAT_RGBA  *ToARGBMatrix   matrixYVU
+    // AVIF_RGB_FORMAT_ABGR  n/a             n/a
+    // AVIF_RGB_FORMAT_565   n/a             n/a
+
+    // Look up table for YUV To RGB Matrix (without filter).
+    typedef int (
+        *YUVToRGBMatrix)(const uint16_t *, int, const uint16_t *, int, const uint16_t *, int, uint8_t *, int, const struct YuvConstants *, int, int);
+    YUVToRGBMatrix lutYuvToRgbMatrix[RGB_PIXEL_FORMATS][YUV_PIXEL_FORMATS] = {
+        {},                                                                   // RGB
+        { NULL, I410ToARGBMatrix, I210ToARGBMatrix, I010ToARGBMatrix, NULL }, // RGBA
+        {},                                                                   // ARGB
+        {},                                                                   // BGR
+        { NULL, I410ToARGBMatrix, I210ToARGBMatrix, I010ToARGBMatrix, NULL }, // BGRA
+        {},                                                                   // ABGR
+        {},                                                                   // RGB_565
+    };
+
+    // Look up table for YUV To RGB Matrix (with filter).
+    typedef int (*YUVToRGBMatrixFilter)(const uint16_t *,
+                                        int,
+                                        const uint16_t *,
+                                        int,
+                                        const uint16_t *,
+                                        int,
+                                        uint8_t *,
+                                        int,
+                                        const struct YuvConstants *,
+                                        int,
+                                        int,
+                                        enum FilterMode);
+    YUVToRGBMatrixFilter lutYuvToRgbMatrixFilter[RGB_PIXEL_FORMATS][YUV_PIXEL_FORMATS] = {
+        {},                                                                   // RGB
+        { NULL, NULL, I210ToARGBMatrixFilter, I010ToARGBMatrixFilter, NULL }, // RGBA
+        {},                                                                   // ARGB
+        {},                                                                   // BGR
+        { NULL, NULL, I210ToARGBMatrixFilter, I010ToARGBMatrixFilter, NULL }, // BGRA
+        {},                                                                   // ABGR
+        {},                                                                   // RGB_565
+    };
+
+    // Look up table for isYVU.
+    avifBool lutIsYVU[RGB_PIXEL_FORMATS] = {
```

Comment:
I am surprised that this `lutIsYVU` table is different from the other `lutIsYVU` table. Why?

Can we make these two `lutIsYVU` tables the same so that the table can be declared as const and shared by the two functions?

Reply:
actually these two tables are the same. in the 10bit case we only use two values (RGBA and BGRA) which are the same as the table above and the other values are unused (and being set to false in this table since it doesn't matter).

But yes, sharing this table makes sense. Done.